File: string/937/937.js
// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare

var reorderLogFiles = function(logs) {
  const getBody = s => s.slice(s.indexOf(" ") + 1);
  const isDigit = s => /\d/.test(s);
  const letterCompare = (a, b) => {
    const n = getBody(a).localeCompare(getBody(b));
    if (n !== 0) return n;
    else return a.localeCompare(b);
  };
  let letterLogs = [];
  let digitLogs = [];
  for (const log of logs) {
    if (isDigit(getBody(log))) digitLogs.push(log);
    else letterLogs.push(log);
  }
  return [...letterLogs.sort(letterCompare), ...digitLogs];
};

File: string/937/937.py
class Solution:
    def reorderLogFiles(self, logs: List[str]) -> List[str]:
        digits = []
        letters = []
        for log in logs:
            if log.split()[1].isdigit():
                digits.append(log)
            else:
                letters.append(log)

        # if there's a tie, sort by identifier
        letters.sort(key=lambda log: log.split()[0])
        # if not sort by log body
        letters.sort(key=lambda log: log.split()[1:])

        return letters+digits
